pkg/upgrade/upgrade_state_test.go

```diff
@@ -1365,9 +1374,127 @@ var _ = Describe("UpgradeStateManager tests", func() {
 		}).WithTimeout(10 * time.Second).WithPolling(1 * 500 * time.Millisecond).Should(Succeed())
 	})
 
+	It("UpgradeStateManager should move to 'node-maintenance-required' while using upgrade requestor mode"+
+		"and shared-requestor flow", func() {
+
+		namespace := createNamespace(fmt.Sprintf("namespace-%s", id)).Name
+		cancel := withUpgradeRequestorMode(testCtx, namespace, false)
+		defer cancel()
+
+		testRequestorID = SecondaryTestRequestorID
+		clusterState := withClusterUpgradeState(3, upgrade.UpgradeStateUpgradeRequired, namespace, nil, true)
+		policy := &v1alpha1.DriverUpgradePolicySpec{
+			AutoUpgrade: true,
+			DrainSpec: &v1alpha1.DrainSpec{
+				Enable: true,
+			},
+		}
+
+		By("verify generated node-maintenance obj(s)")
+		nms := &maintenancev1alpha1.NodeMaintenanceList{}
+		Eventually(func() bool {
```

Comment:
there is no use of cached client here IIUC. so i dont think any of the Eventually blocks are needed. you can just check after the call to ApplyState